### PR TITLE
Port `CListBox` from upstream, smooth scrolling for all lists

### LINKED ISSUE #5791: Smooth scrolling for all scrollregions
Port `CListBox` from upstream and use it instead of `UiDoListbox`.

### PR DESCRIPTION
**Pull request by Robyt3**
Replace existing listbox implementations (`CMenus::UiDoListbox*` and `HandleListInputs` functions) with `CListBox` from upstream.

Reimplement additional feature that was already present in ddnet: page up/down, home and end key handling.

Affects the following lists:

- server browser
  - see screenshots for scoreboard
- server browser scoreboard
  - before:
server-browser-scoreboard old
  - after:
server-browser-scoreboard new
- server browser friends
  - before:
server-browser-friends old
  - after:
server-browser-friends new
- country / region selection popup (server browser filter)
  - before:
country-popup old
  - after:
country-popup new
- player skin list
  - before:
player-skins old
  - after:
player-skins new
- player country / region list
  - before:
player-country old
  - after:
player-country new
- theme list
  - before:
themes new
  - after:
themes old
- assets list
  - before:
assets old
  - after:
assets new
- graphics resolutions list
  - before:
resolutions-dropdown old
  - after:
resolutions-dropdown new
- dropdown menus (e.g. graphics fullscreen mode)
  - see screenshots for graphics resolutions list
- ingame player list
  - before:
players old
  - after:
players new
  - before (low number): 
players-few old
  - after (low number):
players-few new
- vote options list
  - before:
vote-options old
  - after:
vote-options new
- kick/specvote lists
  - before:
kick-spec-vote old
  - after:
kick-spec-vote new
- ghost list
  - before:
ghosts old
  - after:
ghosts new
- language list (in settings and in popup on first launch)
  - settings:
    - before: 
languages-setting old
    - after:
languages-setting new
  - popup:
    - before:
languages-popup old
    - after:
languages-popup new
- demo browser
  - before:
demos old
  - after:
demos new
- editor file browser (saving, loading, adding images / sounds)
  - before:
editor old
  - after:
editor new
  - The search / filename input is also improved so navigating a filtered list works correctly by porting the logic from upstream.

There are minor changes to the visual appearance of some lists, due to changed margins.

The vertical alignment of some list item texts is improved so the text is centered vertically.

Closes #5791.

## Checklist

- [X] Tested the change ingame
- [X] Provided screenshots if it is a visual change
- [ ] Tested in combination with possibly related configuration options
- [ ] Written a unit test (especially base/) or added coverage to integration test
- [ ] Considered possible null pointers and out of bounds array indexing
- [ ] Changed no physics that affect existing maps
- [ ] Tested the change with [ASan+UBSan or valgrind's memcheck](https://github.com/ddnet/ddnet/#using-addresssanitizer--undefinedbehavioursanitizer-or-valgrinds-memcheck) (optional)

### CODE REVIEW

**Comment by def-** [comment]
Wow, thanks a lot @Robyt3 for this huge modernization and in general the large amount of changes! You've been the most active contributor in DDNet for a while now :)

File: CMakeLists.txt [hunk 1]
```diff
@@ -2277,6 +2277,8 @@ if(CLIENT)
     skin.h
     ui.cpp
     ui.h
+    ui_listbox.cpp
```
Comment (def-) [inline CMakeLists.txt:2280]: after ui_rect

Comment (Robyt3) [reply CMakeLists.txt:2280]: Looks correct to me. If it wasn't alphabetically sorted then the build would fail.

Comment (def-) [reply CMakeLists.txt:2280]: oops :D

**Pull request merged**